packages/node_modules/@webex/webex-core/src/interceptors/user-agent.js

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright (c) 2015-2019 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {Interceptor} from '@webex/http-core';
+import {get} from 'lodash';
+
+const strings = new WeakMap();
+
+/**
+ * Sets a 'user-agent' header on all requests if one is not present.
+ * Defaults to '@webex/http-core' though a custom user-agent can be set
+ * using the appName and appVersion configuration. e.g.
+ *
+ *   webex = WebexSdk.init({
+ *     credentials: {
+ *       supertoken: superToken
+ *     },
+ *     config: {
+ *       credentials: {
+ *         client_id,
+ *         client_secret
+ *       },
+ *       appName: 'custom-user-agent',
+ *       appVersion: '1.0',
+ *     }
+ *   });
+ */
+export default class UserAgentInterceptor extends Interceptor {
+  /**
+   * @param {Object} [options={}]
+   * @param {WebexCore} [options.webex]
+   * @private
+   * @returns {UserAgentInterceptor}
+   */
+  constructor(options = {}) {
+    const appName = get(options, 'webex.config.appName');
+    const appVersion = get(options, 'webex.config.appVersion') || '0.0';
+
+    super(options);
+    if (appName) {
+      strings.set(this, `${appName}/${appVersion}`);
+    }
+    else {
+      strings.set(this, '@webex/http-core'); // Using the traditional default from http-core
+    }
+  }
+
+  /**
+   * @returns {UserAgentInterceptor}
+   */
+  static create() {
+    return new UserAgentInterceptor({webex: this});
+  }
+
+  /**
+   * @see Interceptor#onRequest
+   * @param {Object} options
+   * @returns {Object}
+   */
+  onRequest(options) {
+    options.headers = options.headers || {};
+
+    if ('user-agent' in options.headers && options.headers['spark-user-agent']) {
+      return options;
+    }
+
+    options.headers['user-agent'] = strings.get(this);
```

Comment:
This value will be parsed by the Metric service and set in influxdb as **uatype** and **uaversion**. e.g.

> user-agent == ediscovery-downloader/1.0.0

will add the _tag_ **uatype** == 'ediscovery-downloader' and the _field_ **uaversion** == '1.0.0' to InfluxDB.